packages/discord.js/src/structures/Embed.js

```diff
@@ -1,11 +1,195 @@
 'use strict';
 
-const { Embed: BuildersEmbed } = require('@discordjs/builders');
-const Transformers = require('../util/Transformers');
+const isEqual = require('fast-deep-equal');
 
-class Embed extends BuildersEmbed {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+class Embed {
+  /**
+   * Creates a new embed object
+   * @param {APIEmbed} data API embed data
+   * @private
+   */
+  constructor(data = {}) {
+    /**
+     * The API embed data
+     * @type {APIEmbed}
+     * @readonly
+     */
+    this.data = { ...data };
+  }
+
+  /**
+   * An array of fields of this embed
+   * @type {?Array<APIEmbedField>}
+   * @readonly
+   */
+  get fields() {
+    return this.data.fields;
+  }
+
+  /**
+   * The embed title
+   * @type {?string}
+   * @readonly
+   */
+  get title() {
+    return this.data.title;
+  }
+
+  /**
+   * The embed description
+   * @type {?string}
+   * @readonly
+   */
+  get description() {
+    return this.data.description;
+  }
+
+  /**
+   * The embed URL
+   * @type {?string}
+   * @readonly
+   */
+  get url() {
+    return this.data.url;
+  }
+
+  /**
+   * The embed color
+   * @type {?number}
+   * @readonly
+   */
+  get color() {
+    return this.data.color;
+  }
+
+  /**
+   * The timestamp of the embed in an ISO 8601 format
+   * @type {?string}
+   * @readonly
+   */
+  get timestamp() {
+    return this.data.timestamp;
+  }
+
+  /**
+   * The embed thumbnail data
+   * @type {?EmbedImageData}
+   * @readonly
+   */
+  get thumbnail() {
+    if (!this.data.thumbnail) return undefined;
+    return {
+      url: this.data.thumbnail.url,
+      proxyURL: this.data.thumbnail.proxy_url,
+      height: this.data.thumbnail.height,
+      width: this.data.thumbnail.width,
+    };
+  }
+
+  /**
+   * The embed image data
+   * @type {?EmbedImageData}
+   * @readonly
+   */
+  get image() {
+    if (!this.data.image) return undefined;
+    return {
+      url: this.data.image.url,
+      proxyURL: this.data.image.proxy_url,
+      height: this.data.image.height,
+      width: this.data.image.width,
+    };
+  }
+
+  /**
+   * Received video data
+   * @type {?EmbedVideoData}
+   * @readonly
+   */
+  get video() {
+    return this.data.video;
+  }
+
+  /**
+   * The embed author data
+   * @type {?EmbedAuthorData}
+   * @readonly
+   */
+  get author() {
+    if (!this.data.author) return undefined;
+    return {
+      name: this.data.author.name,
+      url: this.data.author.url,
+      iconURL: this.data.author.icon_url,
+      proxyIconURL: this.data.author.proxy_icon_url,
+    };
+  }
+
+  /**
+   * Received data about the embed provider
+   * @type {?EmbedProvider}
+   * @readonly
+   */
+  get provider() {
+    return this.data.provider;
+  }
+
+  /**
+   * The embed footer data
+   * @type {?EmbedFooterData}
+   * @readonly
+   */
+  get footer() {
+    if (!this.data.footer) return undefined;
```

Comment:
explicit undefined is never a good idea, I think this would be better
```suggestion
    if (!this.data.footer) return this.data.footer;
```

Reply:
Can you explain why explicit `undefined` is never a good idea?

Reply:
It’s just general bad practice and there’s linting rules to disallow it. I think you should even return null unless there’s a problem with that